File: usr/share/oss/tools/oss_squid_sso.pl
#!/usr/bin/perl

$| = 1; # do not buffer stdout

while(<STDIN>) {
        chomp;
        my $user = `curl -sX GET --header 'Content-Type: application/json' --header 'Accept: text/plain' http://localhost:9080/api/devices/loggedIn/$_`;
        if ( "$user" ) {
                print "OK user=\"$user\"\n";
        }
        else
        {
                print "OK user=\"default user\"\n";
        }
}

File: usr/share/oss/tools/oss_squid_sso.py
#!/usr/bin/python3

import requests

while True:
    ip=input("")
    user = requests.get('http://localhost:9080/api/devices/loggedIn/' + ip.strip()).text
    if user == "":
        print('OK user="default_user"')
    else:
        print('OK user="' + user + '"')

